first draft on a page to submit bug reports (#300)
humanmade/altis-documentation: other-docs/guides/how-to-submit-a-bug-report.md

# How to submit a bug report.

> Bugs happen, and we’re here to support you, should you find a bug in Altis.

Reviewer: ```suggestion
Bugs happen, and we’re here to support you. If you find a bug in Altis, we'll handle fixing it.
```

> To get the most out of support and to help our product team, there are some requirements for a bug report that must be met, in order for us to help you with your issue.

Reviewer: ```suggestion
To help our product team understand your issue, there are some requirements for a bug report that must be met. This allows our team to know how to both identify the problem, and whether it has been resolved.
```

> When filing a bug report ( [you can read this guide on how to file support tickets](https://docs.altis-dxp.com/nightly/guides/getting-help-with-altis/) ) please provide all of the following:

Reviewer: ```suggestion
When [filing a bug report](./getting-help-with-altis.md) please provide all of the following:
```

## Test Results.

Please write a brief description of the bug, including what you expect to happen and what is currently happening.

> ## [Reduced Test Case](https://css-tricks.com/reduced-test-cases/).
> 
> Please ensure you test the bug in as a reduced test case, and provide the reduced test case as the basis of the bug report.

Reviewer: ```suggestion

## Reduced Test Case

Bugs should be isolated to a [reduced test case](https://css-tricks.com/reduced-test-cases/), as this helps us to ensure that the bug exists in Altis, rather than your project. Reduced test cases should be created against a standard Altis local environment with just the minimal changes needed to reproduce the bug.

Please ensure you provide the reduced test case as the basis of the bug report. If we cannot confirm that the test case is within Altis, we may ask you to reduce the test case further to ensure it isn't caused by your custom code.

Customers on Enterprise support plans may wish to use their dedicated technical engineering support to handle bug isolation.

```

## Steps to reproduce.

> Detailed step-by-step instructions to reproduce the issue.
> ## Environment Setup and Configuration.

Reviewer: ```suggestion
If possible, please also provide detailed step-by-step instructions to reproduce the issue. This will ensure that we can replicate the problem, and avoids back-and-forth conversations.

## Environment Setup and Configuration.
```

> # How to submit a bug report.

Reviewer 2: ```suggestion
# How to Submit a Bug Report
```

Bugs happen, and we’re here to support you. If you find a bug in Altis, we'll handle fixing it.

To help our product team understand your issue, there are some requirements for a bug report that must be met. This allows our team to know how to both identify the problem, and whether it has been resolved.

When [filing a bug report](./getting-help-with-altis.md) please provide all of the following:

> ## Test Results.

Reviewer 2: ```suggestion
## Test Results
```

Please write a brief description of the bug, including what you expect to happen and what is currently happening.

## Reduced Test Case

Bugs should be isolated to a [reduced test case](https://css-tricks.com/reduced-test-cases/), as this helps us to ensure that the bug exists in Altis, rather than your project. Reduced test cases should be created against a standard Altis local environment with just the minimal changes needed to reproduce the bug.

> Please ensure you provide the reduced test case as the basis of the bug report. If we cannot confirm that the test case is within Altis, we may ask you to reduce the test case further to ensure it isn't caused by your custom code.

Reviewer 2: ```suggestion
Please ensure you provide the reduced test case as the basis of the bug report. If we cannot confirm that the test case is within Altis, we may ask you to reduce the test case further to ensure it isn't caused by your custom code.

The reduced test case can be provided as a code snippet in the report itself or as a link to a [GitHub gist](https://gist.github.com/) if there are multiple files involved.
```

Customers on Enterprise support plans may wish to use their dedicated technical engineering support to handle bug isolation.

## Steps to Reproduce

If possible, please also provide detailed step-by-step instructions to reproduce the issue. This will ensure that we can replicate the problem, and avoids back-and-forth conversations.

> ## Environment Setup and Configuration.

Reviewer 2: ```suggestion
## Environment Setup and Configuration
```

> Please let us know what environments the bug is occurring in, and whether you were able to reproduce the bug locally or not.

Reviewer 2: ```suggestion
Please let us know what environments the bug is occurring in, what configuration you have in `composer.json` under the `extra.altis` property (making sure to redact any sensitive information like API keys), and whether you were able to reproduce the bug locally or not.
```
